### ISSUE TITLE
Some apps are not seen

### ISSUE THREAD

**Issue by Briseur** [humdingerb/quicklaunch#41]
I installed Powermanga in a VM to check if a bug that I reported long ago was still present in actual nighly (hrev56385) and the game doesn't appear in QuickLaunch even after a reboot.
I think that it might come from the executable that has no attributes. If I'm right any other app that have a binary of same kind won't be seen either. Hopefully, they are not so many.

**Comment by humdingerb** [comment]
Yeah, that cannot be helped, see [Tips & Tricks](https://htmlpreview.github.io/?https://github.com/humdingerb/quicklaunch/master/ReadMe.html#tips) (last 2 items) of the documentation.

Resources with application signatures (plus icon and possible supprted file formats) can be easily added when an app is packaged. You may want to consider opening tickets at haikuports to add a resource file to an app.
Or even submit PRs for those apps, there are many [examples](https://xref.landonf.org/source/search?q=rdef.in&defs=&refs=&path=&hist=&type=&project=haikuports) showing how it's done. :)

**Comment by ChaplinTokyo** [comment]
Ah... is this the reason apps whose names have been translated into Japanese like Screensaver (=スクリーンセーバー) and Calculator(=電卓）can only be found by their English names, and not by the translated names?  ie. because the app's resource files remain unchanged even when the interface language is translated?

**Comment by humdingerb** [comment]
It's not the resources not being translated - though this would be another issue if QuickLaunch will ever provide the option to search and display the "short/long description" of an app, which is in those resources too.

No, in case of translated app names, it has to do with QuickLaunch's simple (simplistic?) use of a query to find all applications with a certain name. The "name" attribute that's queryable is always the English original.
I've planned for quite some time to rewrite that part. It would be less elegant and possibly needs more code to deal with newly un/installed apps etc. I'd query for all applications and put them all in one big list, then go through that list and request the translated name for the current locale. Then I search through that list when the user inputs a search term.
That's the blurry plan at the moment. Maybe when I find motivation and time over the winter... We'll see. :)

**Comment by humdingerb** [comment]
The solution was easier than I expected... With the latest commits, localized app names should be supported.

@ChaplinTokyo would you like to give it a test run to see if it all works for you?
This is a 64bit built on Haiku Beta5: [QuickLaunch.zip](https://github.com/user-attachments/files/17349100/QuickLaunch.zip)

**Comment by humdingerb** [comment]
I'm not quite sure what I'm seeing... :)
Is it that the search isn't initiated with every inserted symbol but only after you remove some of the symbols?

**Comment by ChaplinTokyo** [comment]
When I enter say the Japanese name for Calculator which is 電卓(pronounced dentaku), I type in ‘dentaku’ on the alphanumeric keyboard with the IME activated (in my case I use mozc because I couldn’t get BeCJK to work properly).  The IME converts the alphanumeric input to hiragana でんたく(the four hiragana characters representing the sounds ‘de-n-ta-ku’) on the fly which I then convert to kanji characters 電卓 with a click of the space bar which activates the kanji conversion to the two Chinese derived characters pronounced ‘den-taku’ meaning ‘electric-table’).  If there are multiple words with the same pronunciation the IME provides a dropdown many of the choices from which the user highlights the required one with the arrow key and then selects it with the enter key.  Now, one would expect QuickLaunch to match this up with the app of the same name (‘電卓‘) at this point but it doesn’t.  I guess that the input from the IME into the search field does not get recognized as a key input or something and so incremental search does not activate.  So what I did was to input a space after the search word and the. Deleted the space.  At this point the search was activated and the app 電卓 was found by QuickLaunch.  I guess that because the delete key is a single byte direct input from the keyboard this was able to trigger incremental search whereas the input of the space via the IME was not so recognized because it was a double byte ‘Japanese’ space?iPhoneから送信2024/10/13 0:16、humdinger ***@***.***>のメール:﻿
I'm not quite sure what I'm seeing... :)
Is it that the search isn't initiated with every inserted symbol but only after you remove some of the symbols?

**Comment by humdingerb** [comment]
Thanks for that explanation! Please have a look at this new version: 
[QuickLaunch.zip](https://github.com/user-attachments/files/17353383/QuickLaunch.zip)

I have change it so that if you press any function key,e.g. F1, the filtering is triggered.
Would this be acceptable? Or is another key better suited? I tried with TAB, but that's used by the IME apparently.

I also saw that QuickLaunch being "modal", i.e. preventing any other window to get focus makes it difficult when using the spacebar to summon the IME's selection window. I've now changed the mode from "modal" to "floating". This keeps the QuickLaunch window on top, but allows interacting with other windows.
I'll have to test it a bit, if this change is overall a good one, but I think it is...

**Comment by humdingerb** [comment]
OK, please forget the above... :)
Here's the latest - and last, let's hope - change: 
[QuickLaunch.zip](https://github.com/user-attachments/files/17354994/QuickLaunch.zip)

I've kept the "floating" window so the IME's selection window won't be obscured. But no F-keys are needed.

**Comment by humdingerb** [comment]
Hmmm... I've tested the last version above and it seemed to work fine. Now I always get this blue symbol at the end:

![screenshot1](https://github.com/user-attachments/assets/f7b3daba-4279-44a0-a865-dee36c55a247)

The fact that it's with any app, here StyledEdit, gives me hope that not QuickLaunch's fault...

**Comment by ChaplinTokyo** [comment]
No need to worry about the blue symbol.  I think you probably typed in dentiku 電池区 instead of dentaku 電卓 so that the IME was wondering about whether the 区 is really what you wanted.iPhoneから送信2024/10/13 18:17、humdinger ***@***.***>のメール:﻿
Hmmm... I've tested the last version above and it seemed to work fine. Now I always get this blue symbol at the end:
screenshot1.png (view on web)
The fact that it's with any app, here StyledEdit, gives me hope that not QuickLaunch's fault...

**Comment by humdingerb** [comment]
Oh, duh! :)
In my feable defense, I have a friend with the name "Dentico" and my mind was side-tracked... :)
